src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -62,6 +62,10 @@ public static partial class GC
         [MethodImpl(MethodImplOptions.InternalCall)]
         private static extern void GetMemoryInfo(GCMemoryInfoData data, int kind);
 
+        // Used to avoid allocating in GetGCMemoryInfo after the first call.
```

Comment:
This is a breaking change. The instance is returned to the user and it is expected to be immutable. I do not think we want to do this.